kalliope/core/Models/MatchedSynapse.py

```diff
@@ -28,9 +28,8 @@ def __init__(self, matched_synapse=None, matched_order=None, user_order=None, ov
             self.parameters = NeuronParameterLoader.get_parameters(synapse_order=self.matched_order,
                                                                    user_order=user_order)
         if overriding_parameter is not None:
-            # we suppose that we don't have any parameters.
-            # We replace the current parameter object with the received one
-            self.parameters = overriding_parameter
+            # merge dict of parameters with overriding
+            self.parameters = self.merge_two_dicts(self.parameters, overriding_parameter)
```

Comment:
method merge_two_dicts is useless.
Instead you can use directly : self.parameters.update(overriding_parameter)